node/src/components/contract_runtime.rs

```diff
@@ -858,12 +862,21 @@ impl ContractRuntime {
         engine_state: &EngineState<LmdbGlobalState>,
         metrics: &Metrics,
         trie_or_chunk_id: TrieOrChunkId,
-    ) -> Result<Option<TrieOrChunk>, engine_state::Error> {
+    ) -> Result<Option<TrieOrChunk>, ContractRuntimeError> {
         let correlation_id = CorrelationId::new();
         let start = Instant::now();
-        let result = engine_state.get_trie(correlation_id, trie_or_chunk_id);
-        metrics.get_trie.observe(start.elapsed().as_secs_f64());
-        result
+        let TrieOrChunkId(chunk_index, trie_key) = trie_or_chunk_id;
+        match engine_state.get_trie_full(correlation_id, trie_key)? {
+            None => {
+                metrics.get_trie.observe(start.elapsed().as_secs_f64());
+                Ok(None)
+            }
+            Some(trie_raw) => {
+                let trie_or_chunk = TrieOrChunk::new(trie_raw, chunk_index)?;
+                metrics.get_trie.observe(start.elapsed().as_secs_f64());
+                Ok(Some(trie_or_chunk))
+            }
+        }
```

Comment:
Could this work?
Rationale: try to update the metric in a single place.

``` suggestion
        let ret = match engine_state.get_trie_full(correlation_id, trie_key)? {
            None => Ok(None),
            Some(trie_raw) => Ok(Some(TrieOrChunk::new(trie_raw, chunk_index)?)),
        };
        metrics.get_trie.observe(start.elapsed().as_secs_f64());
        ret
```

Reply:
Done in [`e8243e8` (#3255)](https://github.com/casper-network/casper-node/pull/3255/commits/e8243e8b08580a6e72cebdfed0b1b1957f46652a)